From 999e1159f439949cfe14a2011c94943c7aaa633d Mon Sep 17 00:00:00 2001
From: California Sullivan <california.l.sullivan@intel.com>
Date: Thu, 19 Nov 2015 15:37:34 -0800
Subject: [PATCH] xml2-config is disabled. Use pkg-config instead.

Upstream-Status: total hack
Signed-off-by: California Sullivan <california.l.sullivan@gmail.com>
---
 configure | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/configure b/configure
index 4242042..5f0d36f 100755
--- a/configure
+++ b/configure
@@ -10220,7 +10220,12 @@ if test "x$XML2CONFIG" = "x"; then
  { echo "$as_me:$LINENO: result: xml2-config not found" >&5
 echo "${ECHO_T}xml2-config not found" >&6; }
 else
- LIBXML2_INC=`$XML2CONFIG --cflags`
+ if pkg-config --exists libxml-2.0 ; then
+  echo "using pkg-config instead of xml2-config"
+  LIBXML2_INC=`pkg-config libxml-2.0 --cflags`
+ else
+  LIBXML2_INC=`$XML2CONFIG --cflags`
+ fi
  { echo "$as_me:$LINENO: result: $LIBXML2_INC" >&5
 echo "${ECHO_T}$LIBXML2_INC" >&6; }
  { echo "$as_me:$LINENO: checking for xmlReadFile in -lxml2" >&5
-- 
2.1.0

